File: server/api/src/main/java/com/boxtrotstudio/ghost/common/network/packet/ChangeItemPacket.java
package com.boxtrotstudio.ghost.common.network.packet;

import com.boxtrotstudio.ghost.common.network.BasePlayerClient;
import com.boxtrotstudio.ghost.common.network.BaseServer;
import com.boxtrotstudio.ghost.game.match.LiveMatchImpl;
import com.boxtrotstudio.ghost.network.packet.Packet;

import java.io.IOException;

import static com.boxtrotstudio.ghost.utils.Global.RANDOM;

public class ChangeItemPacket extends Packet<BaseServer, BasePlayerClient> {
    @Override
    public void onHandlePacket(BasePlayerClient client) throws IOException {
        byte action = consume(1).asByte();

        if (client.getPlayer() == null || client.getPlayer().isInMatch())
            return;


        if (action == 0x10 || action < 0 || action >= LiveMatchImpl.ITEMS.length) {
            client.getPlayer().setPreferredItem(RANDOM.nextInt(LiveMatchImpl.ITEMS.length));
        } else {
            client.getPlayer().setPreferredItem(action);
        }
    }
}

File: client/core/src/com/boxtrotstudio/ghost/client/network/packets/ChangeItemPacket.kt
package com.boxtrotstudio.ghost.client.network.packets

import com.boxtrotstudio.ghost.client.network.Packet
import com.boxtrotstudio.ghost.client.network.PlayerClient

class ChangeItemPacket : Packet<PlayerClient>() {

    override fun write(vararg args : Any) {
        val item = args[0] as Byte

        write(0x50.toByte())
        write(item)
        endTCP()
    }
}
